### PR TITLE
Check if the connection is closed before opening a new stream.

### PR DESCRIPTION
**Pull request by Stebalien**
fixes ipfs/go-ipfs#4129

### CODE REVIEW

**Comment by Stebalien** [comment]
"Dead programs don't leak."

File: multiplex.go [hunk 1]
```diff
@@ -167,6 +167,13 @@ func (mp *Multiplex) NewStream() (*Stream, error) {
 
 func (mp *Multiplex) NewNamedStream(name string) (*Stream, error) {
 	mp.chLock.Lock()
+
+	// We could call IsClosed but this is faster (given that we already have
+	// the lock).
+	if mp.channels == nil {
```
Comment (Stebalien) [inline multiplex.go]: @whyrusleeping this is faster (who cares?) but also a bit less reliable (if someone decides to *not* null this out, we'll have problems). Thoughts?

Comment (whyrusleeping) [inline multiplex.go]: why would someone decide not to null this out?

Comment (Stebalien) [inline multiplex.go]: Meh. I've just commented it and left it as-is.

There's no reason to but the fact that `channels == nil` means the connection is closed is non-obvious. I'm probably just paranoid.

**Pull request merged**